src/Microsoft.CorrelationVector/CorrelationVectorV1.cs

```diff
@@ -0,0 +1,404 @@
+                if (p > 0)
+                {
+                    string extensionValue = immutable ? correlationVector.Substring(p + 1, correlationVector.Length - p - 1 - CorrelationVectorV1.TerminationSign.Length)
+                        : correlationVector.Substring(p + 1);
+                    int extension;
+                    if (int.TryParse(extensionValue, out extension) && extension >= 0)
+                    {
+                        return new CorrelationVectorV1(correlationVector.Substring(0, p), extension, CorrelationVectorV1.InferVersion(correlationVector, false), immutable);
+                    }
+                }
+            }
+
+            return new CorrelationVectorV1();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CorrelationVectorV1"/> class. This
+        /// should only be called when no correlation vector was found in the message
+        /// header.
+        /// </summary>
+        public CorrelationVectorV1()
+            : this(CorrelationVectorVersion.V1)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CorrelationVectorV1"/> class of the
+        /// given implemenation version. This should only be called when no correlation
+        /// vector was found in the message header.
+        /// </summary>
+        /// <param name="version">The correlation vector implemenation version.</param>
+        public CorrelationVectorV1(CorrelationVectorVersion version)
+            : this(CorrelationVectorV1.GetUniqueValue(version), 0, version, false)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CorrelationVectorV1"/> class of the
+        /// V2 implemenation using the given <see cref="System.Guid"/> as the vector base.
+        /// </summary>
+        /// <param name="vectorBase">The <see cref="System.Guid"/> to use as a correlation
+        /// vector base.</param>
+        public CorrelationVectorV1(Guid vectorBase)
+            : this(vectorBase.GetBaseFromGuid(), 0, CorrelationVectorVersion.V2, false)
+        {
+        }
+
+        /// <summary>
+        /// Gets the value of the correlation vector as a string.
+        /// </summary>
+        public override string Value
+        {
+            get
+            {
+                return string.Concat(this.BaseVector, ".", this.extension,
+                    this.immutable ? CorrelationVectorV1.TerminationSign : string.Empty);
+            }
+        }
+
+        /// <summary>
+        /// Increments the current extension by one. Do this before passing the value to an
+        /// outbound message header.
+        /// </summary>
+        /// <returns>
+        /// The new value as a string that you can add to the outbound message header
+        /// indicated by <see cref="HeaderName"/>.
+        /// </returns>
+        public override string Increment()
```

Comment:
You can move this logic into the abstract CorrelationVector since its going to be the same for both V1 and V2

Reply:
the problem is that Increment will be very different for V3.

Reply:
we will be overriding it in V3 and we can keep the method as virtual in CorrelationVector (so that it can be overriden). It just helps with reducing duplicate code

Reply:
The problem with doing that is that I don't want to add the "immutable" property to the base cV, which exists as a property in cV v1 and v2. However, it is part of the Increment() method in both v1 and v2.

Reply:
This is because cV v3 isn't supposed to have the immutable property, as any values in v3 are supposed to be reset when they're too big.